### ISSUE TITLE
DateTimeOffset parsing problems in complex objects

### ISSUE THREAD

**Issue by Tornhoof** [neuecc/Utf8Json#20]
Hi, there are still some problems with DTO parsing in complex object. I think this only happens if no sub-second precision values are available. Exception is:
``---- System.InvalidOperationException : invalid datetime format. value:1970-01-01T09:15:10+00:00``

```csharp
using System;
using Utf8Json.Resolvers;

namespace ConsoleApp9
{
    class Program
    {
        static void Main(string[] args)
        {
            var input = "{\"Flag\":false,\"StatusDate\":\"1970-01-01T09:15:10+00:00\"}";
            var deserialized = Utf8Json.JsonSerializer.NonGeneric.Deserialize(typeof(Test), input, StandardResolver.ExcludeNull);
        }
    }

    public class Test
    {
        public bool Flag { get; set; }
        public DateTimeOffset? StatusDate { get; set; }
    }
}

```

**Comment by neuecc** [comment]
Thank you, this is miss of range-check code.
If exists offset, can not parse.
I'll fix it soon.

**Comment by neuecc** [comment]
I've released `v1.2.2`.
Sorry, this is critical issue, my huge mistake.

**Comment by Tornhoof** [comment]
Thanks, it works now, closing this.
